source/lru_cache.c

```diff
@@ -100,16 +107,18 @@ static void *s_lru_cache_get_mru_element(const struct aws_cache *cache) {
 
 void *aws_lru_cache_use_lru_element(struct aws_cache *cache) {
     AWS_ASSERT(cache);
```

Comment:
Probably want to:
```c
AWS_PRECONDITION(cache);
AWS_PRECONDITION(cache->impl);
```
For both of these functions

Reply:
Got it. Should the APIs for aws_cache also do the similar thing?
`int aws_cache_find(struct aws_cache *cache, const void *key, void **p_value) {
    AWS_ASSERT(cache);
    return cache->vtable->find(cache, key, p_value);
}`
to
`int aws_cache_find(struct aws_cache *cache, const void *key, void **p_value) {
    AWS_PRECONDITION(cache);
    return cache->vtable->find(cache, key, p_value);
}`

To be honest, I don't know what's the difference between AWS_PRECONDITION and AWS_ASSERT....